test/util/test_path_provider.py

```diff
@@ -0,0 +1,173 @@
+            self.HELPERS_FILES_DIR = helpers_file_env
+        else:
+            self.HELPERS_FILES_DIR = helpers_file_dir
+
+        if TestRunnerPathProvider.__instance is not None:
+            raise Exception("This class is a singleton!")
+        else:
+            TestRunnerPathProvider.__instance = self
+
+    @property
+    def test_conjurrc_file_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR, "conjurrc")
+
+    @property
+    def test_no_cert_conjurrc_file_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR, "test_config", "no_cert_conjurrc")
+
+    @property
+    def test_bad_cert_conjurrc_file_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR, "test_config", "bad_cert_conjurrc")
+
+    @property
+    def helpers_files_path(self):
+        return self.HELPERS_FILES_DIR
+
+    @property
+    def nginx_conf_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR, "test_config", "https", "nginx.conf")
+
+    def get_policy_path(self, type: str):
+        if type.lower() == "initial":
+            return os.path.join(self.HELPERS_FILES_DIR, "test_config", r"initial_policy.yml")
+
+
+class TestIntegrationPathProvider(TestPathProviderBase):
+    """
+    used to generate the files path's in all tests
+    in particular this file is helping with the integration between running test_integration script
+    and the integration_test_runner.
+    all test path's should go through here.
+    in the future should use the constants module to be more stable
+    """
+    __instance = None
+
+    @staticmethod
+    def getInstance():
+        """
+        Implement the singelton design pattern. so we will be sure all path's are stable
+        @return:  TestPathProvider
+        """
+
+        if TestIntegrationPathProvider.__instance is None:
+            TestIntegrationPathProvider()
+        return TestIntegrationPathProvider.__instance
+
+    def __init__(self):
+        super(TestIntegrationPathProvider, self).__init__(Path.home(), './test')
+
+        if TestIntegrationPathProvider.__instance is not None:
+            raise Exception("This class is a singleton!")
+        else:
+            TestIntegrationPathProvider.__instance = self
+
+    @property
+    def test_conjurrc_file_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR,"test_config", "conjurrc")
+
+    @property
+    def test_no_cert_conjurrc_file_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR, "test_config", "no_cert_conjurrc")
+
+    @property
+    def test_bad_cert_conjurrc_file_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR, "test_config", "bad_cert_conjurrc")
+
+    @property
+    def helpers_files_path(self):
+        return self.HELPERS_FILES_DIR
+
+    @property
+    def nginx_conf_path(self):
+        return os.path.join(self.HELPERS_FILES_DIR, "test_config", "https", "nginx.conf")
+
+    def get_policy_path(self, type: str):
+        if type.lower() == "initial":
```

Comment:
what if there are more policies? do we have to keep adding ifs?
Can we make this method general and maybe have a convention for policies so that we can abstract this method for example:
get_policy_path will take the name (initial) and we would just append `_policy.yml`
that way, when we want to create more policies, we will follow the same convention <>_policy.yml and have an general method to that be reused no matter how many policies we require

Reply:
nice. changing